crates/p2p/src/swarm/handle.rs

```diff
@@ -156,6 +156,33 @@ impl CommandHandle {
             _ => Vec::new(), // Timeout or channel closed
         }
     }
+
+    /// Gets the app public key for a specific peer.
+    /// Returns None if we are not connected to the peer or key exchange hasn't happened yet.
+    pub async fn get_app_public_key(&self, peer_id: PeerId) -> Option<PublicKey> {
+        let (sender, receiver) = oneshot::channel();
+
+        // Send the command.
+        let cmd = Command::QueryP2PState(QueryP2PStateCommand::GetAppPublicKey {
+            transport_id: peer_id,
+            response_sender: sender,
+        });
+
+        // Use a cloned sender to avoid borrow issues.
+        let cmd_sender = self.commands.clone();
+
+        // If sending fails, return None.
+        if cmd_sender.send(cmd).await.is_err() {
+            return None;
+        }
+
+        // Wait for response with timeout.
+        // TODO: make this configurable
```

Comment:
If possible tackle in this PR, or we can always do it later, since there are a lot of PRs waiting for this to be merged.